src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1500,7 +1500,21 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
     else:
         display_diagnostics_report(temp_kubeconfig_path)
 
-    return
+
+def _trim_fqdn_name_containing_hcp(normalized_fqdn: str) -> str:
+    """
+    Trims the storage blob name and takes everything prior to "-hcp-".
+    Currently it is displayed wrong: i.e. at time of creation cli has
+    following limitation:
+    https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/
```

Comment:
To be specific, you are saying that the problems are:
1. the `-` in the name are causing it to fail 
2. having hcp in every name is not ideal as it leads to less characters that are truly unique in the name 

Is that right?

Reply:
@gtracer , the issue here is that the aks-periscope pods will create a blob for MC with name 'xxx-xxx-hcp-xxxx' like this "xxx-xxx". Basically removing everything after "-hcp" (including -hcp). But our CLI code still uses the full MC name in the URL. Hence the mismatch.